### PR TITLE
Automatically trigger the ilasm round-trip test on PRs

### PR DESCRIPTION
**Pull request by BruceForstall**
For changes to the ilasm or ildasm source code, trigger
the ilasm round-trip pipeline.

There could, of course, be other changes that could affect
ilasm/ildasm, but this at least catches the primary ones.

### CODE REVIEW

File: eng/pipelines/coreclr/ilasm.yml [hunk 1]
```diff
@@ -1,5 +1,14 @@
 trigger: none
 
+pr:
```
Comment (BruceForstall) [inline eng/pipelines/coreclr/ilasm.yml:3]: I presume this shouldn't affect anyone's ability to manually trigger the pipeline in any branch for any PR.

**Comment by BruceForstall** [comment]
I think I need to merge this before testing that a PR change to ilasm/ildasm actually triggers this, right? Anyway, it's low cost to do it that way (merge then test).

**Review by trylek — APPROVED** [review]
LGTM

**Comment by BruceForstall** [comment]
@trylek After I merged this, I submitted https://github.com/dotnet/runtime/pull/43673 which changes ildasm, but the ilasm pipeline (https://dev.azure.com/dnceng/public/_build?definitionId=870&_a=summary) didn't get triggered. Can you think of any reasons why? Is there anything else I need to do?

(btw, I'm going to manually trigger it now, so it will run)

**Pull request merged**